xarray/core/dataset.py

```diff
@@ -500,7 +500,7 @@ def __setitem__(self, key, value) -> None:
         self.dataset[dim_indexers] = value
 
 
-class Dataset(
+class Dataset(  # type: ignore[misc,unused-ignore]
```

Comment:
Why is this necessary? Can you post the error message that you get without?
Also, same question: why do you need to add unused-ignore?